src/funnels/automation/StableSwapper.sol

```diff
@@ -0,0 +1,99 @@
+                                                                         // configs[src][dst].lot    [token weis] is the amount swapped by keepers from src to dst every hop
+                                                                         // configs[src][dst].reqOut [token weis] is the minimum output amount to insist on in the swap form src to dst
+
+    address public immutable swapper;                                    // Swapper for this StableSwapper
+
+    event Rely   (address indexed usr);
+    event Deny   (address indexed usr);
+    event Kissed (address indexed usr);
+    event Dissed (address indexed usr);
+    event Permit (address indexed usr);
+    event Forbid (address indexed usr);
+    event File   (bytes32 indexed what, address data);
+    event Config (address indexed src, address indexed dst, PairConfig data);
+
+    constructor(address swapper_) {
+        swapper = swapper_;
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "StableSwapper/not-authorized");
+        _;
+    }
+
+    // permissionned to facilitators
+    modifier toll { 
+        require(buds[msg.sender] == 1, "StableSwapper/non-facilitator"); 
+        _;
+    }
+
+    // permissionned to whitelisted keepers
+    modifier keep { 
+        require(bots[msg.sender] == 1, "StableSwapper/non-keeper"); 
+        _;
+    }
+
+    uint256 internal constant WAD = 10 ** 18;
+
+    function rely  (address usr) external auth { wards[usr] = 1; emit Rely(usr); }
+    function deny  (address usr) external auth { wards[usr] = 0; emit Deny(usr); }
+    function kiss  (address usr) external auth {  buds[usr] = 1; emit Kissed(usr); }
+    function diss  (address usr) external auth {  buds[usr] = 0; emit Dissed(usr); }
+    function permit(address usr) external toll {  bots[usr] = 1; emit Permit(usr); }
+    function forbid(address usr) external toll {  bots[usr] = 0; emit Forbid(usr); }
+
+    struct PairConfig {
+        uint32 count;
+        uint112 lot;
+        uint112 reqOut;
+    }
+
+    function setConfig(address src, address dst, PairConfig memory cfg) external toll {
+        configs[src][dst] = cfg;
+        emit Config(src, dst, cfg);
+    }
+
+    // Note: the keeper's minOut value must be updated whenever configs[src][dst] is changed.
+    // Failing to do so may result in this call reverting or in taking on more slippage than intended (up to a limit controlled by configs[src][dst].reqOut).
+    function swap(address src, address dst, uint256 minOut, address callee, bytes calldata data) keep external returns (uint256 out) {
+        PairConfig memory cfg = configs[src][dst];
+
+        require(cfg.count > 0, "StableSwapper/exceeds-count");
+        configs[src][dst].count = cfg.count - 1;
+
+        if (minOut == 0) minOut = cfg.reqOut;
```

Comment:
Wouldn't just be easier to reason to use the max between both `minOut` and `cfg.reqOut`?

Reply:
The idea was that we don't want to force a different limit that the keeper didn't intend to set.
There is support for the keeper to knowingly rely on the required price that the facilitator set by passing 0.

But I also think it is a bit over-complicated as is, so maybe using the max is fine and we can re-complicate it if needed.

Reply:
ok, let's see what @telome thinks about it

Reply:
I think it's probably better to explicitly revert in case minOut was not correctly adjusted after a change of config. We made it so that all config params need to be changed at once in part to prevent keepers from silently running with out-of-sync params so in the spirit of consistency I would also try to disallow running keepers when minOut is known to be incorrectly set (which would be the case when 0 < minOut < reqOut)